### PR TITLE
202006 release notes created

### PR DESCRIPTION
**Pull request by kannankvs**
Release notes for review

### CODE REVIEW

File: doc/SONiC_202006_Release_Notes.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+1. Kernal upgraded from 4.9.110-3deb9u6 (SONiC Release 201904) to 4.9.168-1+deb9u5 in this SONiC release. 
+   Change log: https://tracker.debian.org/media/packages/l/linux/changelog-4.9.168-1deb9u5
+2. Docker upgraded from 18.09.2\~3-0\~debian-stretch to 18.09.8\~3-0\~debian-stretch. 
+   Change log: https://docs.docker.com/engine/release-notes/#18098 
+
+# Feature List
+
+#### Build Improvements 
+This document provides functional design and specifications of BFD protocol as defined in RFC 5880, 5881, 5882 and 5883.BFD protocol defines a method of rapid detection of the failure of a forwarding path by checking that the next hop router is alive. The protocol will be able to detect the forwarding path failure in milliseconds depending on the actual configuration.
```
Comment (ben-gale) [inline doc/SONiC_202006_Release_Notes.md:51]: This content (BFD) does not match the heading (Build Improvements). The topic here is the latter.

File: doc/SONiC_202006_Release_Notes.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+# Feature List
+
+#### Build Improvements 
+This document provides functional design and specifications of BFD protocol as defined in RFC 5880, 5881, 5882 and 5883.BFD protocol defines a method of rapid detection of the failure of a forwarding path by checking that the next hop router is alive. The protocol will be able to detect the forwarding path failure in milliseconds depending on the actual configuration. 
+<br> Refer [HLD document]() and below mentioned PR's for more details. 
+<br> **Pull Requests** : [3292](https://github.com/Azure/sonic-buildimage/pull/3292), [4117](https://github.com/Azure/sonic-buildimage/pull/4117), [4425](https://github.com/Azure/sonic-buildimage/pull/4425) 
+
+#### Bulk API for route
+This document provides functional design and specifications of BFD protocol as defined in RFC 5880, 5881, 5882 and 5883.BFD protocol defines a method of rapid detection of the failure of a forwarding path by checking that the next hop router is alive. The protocol will be able to detect the forwarding path failure in milliseconds depending on the actual configuration.
```
Comment (ben-gale) [inline doc/SONiC_202006_Release_Notes.md:56]: Same problem - cut and paste errors - pls fix

Comment (ben-gale) [reply doc/SONiC_202006_Release_Notes.md:56]: Also the HLD link just points to the folder - not very helpful - would be better to point to the actual HLD. Similar for the prior section also.

File: doc/SONiC_202006_Release_Notes.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+<br> **Pull Requests** : [1238](https://github.com/Azure/sonic-swss/pull/1238)  
+
+#### D-Bus to Host Communications 
+This document describes a means (framework) for an application executed inside a container to securely request the execution of an operation ("action") by the host OS.This framework is intended to be used by the SONiC management and telemetry containers, but can be extended for other application containers as well.
+<br> Refer [HLD document](https://github.com/Azure/SONiC/blob/master/doc/mgmt/Docker%20to%20Host%20communication.md) and below mentioned PR's for more details. 
+<br> **Pull Requests** : [4840](https://github.com/Azure/sonic-buildimage/pull/4840)
+
+#### Debian 10 upgrade, base image,driver 
+
+<br> Refer [HLD document]() and below mentioned PR's for more details.
```
Comment (ben-gale) [inline doc/SONiC_202006_Release_Notes.md:67]: Please point to the actual HLD (not the doc folder)

File: doc/SONiC_202006_Release_Notes.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+This document describes a means (framework) for an application executed inside a container to securely request the execution of an operation ("action") by the host OS.This framework is intended to be used by the SONiC management and telemetry containers, but can be extended for other application containers as well.
+<br> Refer [HLD document](https://github.com/Azure/SONiC/blob/master/doc/mgmt/Docker%20to%20Host%20communication.md) and below mentioned PR's for more details. 
+<br> **Pull Requests** : [4840](https://github.com/Azure/sonic-buildimage/pull/4840)
+
+#### Debian 10 upgrade, base image,driver 
+
+<br> Refer [HLD document]() and below mentioned PR's for more details. 
+<br> **Pull Requests** : [145](https://github.com/Azure/sonic-linux-kernel/pull/145), [4711](https://github.com/Azure/sonic-buildimage/pull/4711) 
+
+#### Dynamic port break
```
Comment (ben-gale) [inline doc/SONiC_202006_Release_Notes.md:70]: "breakout"

File: doc/SONiC_202006_Release_Notes.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+Ports can be broken out to different speeds with various lanes in most HW today. However, on SONiC, the port breakout modes are hard-coded in the profiles and only loaded at initial time. In case we need to have a new port breakout mode, we would potentially need a new image or at least need to restart services which would impact the traffic of the box on irrelevant ports. The feature is to address the above issues.
```
Comment (ben-gale) [inline doc/SONiC_202006_Release_Notes.md:71]: Suggest " However, on SONiC, before this release, the port breakout modes ...."

File: doc/SONiC_202006_Release_Notes.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+#### Egress shaping (port, queue) 
+Quality of Service (QoS) scheduling and shaping features enable better service to certain traffic flows.Queue scheduling provides preferential treatment of traffic classes mapped to specific egress queues. SONiC supports SP, WRR, and DWRR scheduling disciplines.Queue shaping provides control of minimum and maximum bandwidth requirements per egress queue for more effective bandwidth utilization. Egress queues that exceed an average transmission rate beyond the shaper max bandwidth will stop being serviced. Additional ingress traffic will continue to be stored on the egress queue until the queue size is exceeded which results in tail drop.
+<br> Refer [HLD document](https://github.com/Azure/SONiC/blob/41e55d2762e9267454a4910b42a1eb7ad07acda8/doc/qos/scheduler/SONiC_QoS_Scheduler_Shaper.md) and below mentioned PR's for more details. 
+<br> **Pull Requests** : [1296](https://github.com/Azure/sonic-swss/pull/1296), [991](https://github.com/Azure/sonic-swss/pull/991)
+
+#### FW utils extension: SSD upgrade
```
Comment (ben-gale) [inline doc/SONiC_202006_Release_Notes.md:80]: What does this have to do with SSD upgrade? It's not even mentioned below.

File: doc/SONiC_202006_Release_Notes.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+<br> Refer [HLD document](https://github.com/Azure/SONiC/blob/master/doc/fwutil/fwutil.md) and below mentioned PR's for more details. 
+<br> **Pull Requests** : [4764](https://github.com/Azure/sonic-buildimage/pull/4764), [4758](https://github.com/Azure/sonic-buildimage/pull/4758), [941](https://github.com/Azure/sonic-utilities/pull/941), [942](https://github.com/Azure/sonic-utilities/pull/942), [87](https://github.com/Azure/sonic-platform-common/pull/87), [82](https://github.com/Azure/sonic-platform-common/pull/82)
+
+#### Getting docker ready for Debian 10
+
+<br> Refer [HLD document]() and below mentioned PR's for more details.
```
Comment (ben-gale) [inline doc/SONiC_202006_Release_Notes.md:87]: Please point to the actual HLD

File: doc/SONiC_202006_Release_Notes.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+#### Proxy ARP  
+When an interface is enabled with "proxy_arp", the same is enabled in the kernel. ASIC ARP packet action is also updated to trap these packets to CPU in those interfaces.
+<br> Refer [HLD Document](https://github.com/Azure/SONiC/blob/master/doc/arp/Proxy%20Arp.md) for more details. 
+<br> **Pull Requests** :  [617](https://github.com/Azure/SONiC/pull/617)
+
+#### Pytest 100% moved from ansible to Pytest 
+
+#### SPytest
+
+<br> Refer [HLD Document]() for more details.
```
Comment (ben-gale) [inline doc/SONiC_202006_Release_Notes.md:104]: Doc is here - https://github.com/Azure/sonic-mgmt/blob/master/spytest/Doc/intro.md

**Review by ben-gale — CHANGES_REQUESTED** [review]

File: doc/SONiC_202006_Release_Notes.md [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+| swss                      | 1.0.0    |
+| radvd                     | 2.17-2~bpo9+1    |
+| isc-dhcp                  | 4.3.5-2 ([PR2946](https://github.com/Azure/sonic-buildimage/pull/2946) )   |
+| sonic-telemetry           | 0.1    |
+| redis-server/ redis-tools | 5.0.3-3~bpo9+2    |
+
+
+# Security Updates
+
+1. Kernal upgraded from 4.9.110-3deb9u6 (SONiC Release 201904) to 4.9.168-1+deb9u5 in this SONiC release.
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply doc/SONiC_202006_Release_Notes.md]: Done.

File: doc/SONiC_202006_Release_Notes.md (hunk 1 above)
Comment (kannankvs) [reply doc/SONiC_202006_Release_Notes.md]: Done.

File: doc/SONiC_202006_Release_Notes.md (hunk 2 above)
Comment (kannankvs) [reply doc/SONiC_202006_Release_Notes.md]: Done.

File: doc/SONiC_202006_Release_Notes.md (hunk 3 above)
Comment (kannankvs) [reply doc/SONiC_202006_Release_Notes.md]: Done.

File: doc/SONiC_202006_Release_Notes.md (hunk 4 above)
Comment (kannankvs) [reply doc/SONiC_202006_Release_Notes.md]: Done.

File: doc/SONiC_202006_Release_Notes.md (hunk 5 above)
Comment (kannankvs) [reply doc/SONiC_202006_Release_Notes.md]: Done.

File: doc/SONiC_202006_Release_Notes.md (hunk 6 above)
Comment (kannankvs) [reply doc/SONiC_202006_Release_Notes.md]: Done.

File: doc/SONiC_202006_Release_Notes.md (hunk 7 above)
Comment (kannankvs) [reply doc/SONiC_202006_Release_Notes.md]: Done.

File: doc/SONiC_202006_Release_Notes.md (hunk 8 above)
Comment (kannankvs) [reply doc/SONiC_202006_Release_Notes.md]: Done.

File: doc/SONiC_202006_Release_Notes.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (ben-gale) [reply doc/SONiC_202006_Release_Notes.md]: Ack

File: doc/SONiC_202006_Release_Notes.md (hunk 1 above)
Comment (ben-gale) [reply doc/SONiC_202006_Release_Notes.md]: Ack

File: doc/SONiC_202006_Release_Notes.md (hunk 2 above)
Comment (ben-gale) [reply doc/SONiC_202006_Release_Notes.md]: Ack

File: doc/SONiC_202006_Release_Notes.md (hunk 3 above)
Comment (ben-gale) [reply doc/SONiC_202006_Release_Notes.md]: Ack

File: doc/SONiC_202006_Release_Notes.md (hunk 4 above)
Comment (ben-gale) [reply doc/SONiC_202006_Release_Notes.md]: Ack

File: doc/SONiC_202006_Release_Notes.md (hunk 5 above)
Comment (ben-gale) [reply doc/SONiC_202006_Release_Notes.md]: I guess you omitted the "before this release" part, but fair enough

File: doc/SONiC_202006_Release_Notes.md (hunk 6 above)
Comment (ben-gale) [reply doc/SONiC_202006_Release_Notes.md]: Ack

File: doc/SONiC_202006_Release_Notes.md (hunk 7 above)
Comment (ben-gale) [reply doc/SONiC_202006_Release_Notes.md]: Ack

File: doc/SONiC_202006_Release_Notes.md (hunk 8 above)
Comment (ben-gale) [reply doc/SONiC_202006_Release_Notes.md]: Ack

**Comment by ben-gale** [comment]
Looks good ...

**Review by henry-ma — CHANGES_REQUESTED** [review]

File: doc/SONiC_202006_Release_Notes.md [hunk 10]
```diff
@@ -0,0 +1,134 @@
+# SONiC 201911 Release Notes
```
Comment (bluecmd) [inline doc/SONiC_202006_Release_Notes.md]: s/201911/202006/

File: doc/SONiC_202006_Release_Notes.md [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+#### Pytest 100% moved from ansible to Pytest 
+
+#### SPytest
+
+<br> Refer [HLD Document]() for more details. 
+<br> **Pull Requests** :  [1533](https://github.com/Azure/sonic-mgmt/pull/1533)
+
+#### Thermal control 
+Thermal control daemon has been added to monitor the temperature of devices (CPU, ASIC, optical modules, etc) and the running status of fan. It retrieves the switch device temperatures via platform APIs and raises alarms when the high/low thresholds are hit.It also stores temperature values fetched from sensors and thermal device running status to the DB.
+<br> Refer  [HLD Document](https://github.com/Azure/SONiC/blob/master/thermal-control-design.md) for more details.
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply doc/SONiC_202006_Release_Notes.md]: Addressed it. Pls check and confirm.

File: doc/SONiC_202006_Release_Notes.md [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+<br> **Pull Requests** :  [73](https://github.com/Azure/sonic-platform-common/pull/73), [777](https://github.com/Azure/sonic-utilities/pull/777), [49](https://github.com/Azure/sonic-platform-daemons/pull/49), [3949](https://github.com/Azure/sonic-buildimage/pull/3949),[832](https://github.com/Azure/sonic-utilities/pull/832) 
+
+
+<br>
+
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply doc/SONiC_202006_Release_Notes.md]: Addressed it. Pls check and confirm.

File: doc/SONiC_202006_Release_Notes.md [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply doc/SONiC_202006_Release_Notes.md]: Addressed it. Pls check and confirm.

File: doc/SONiC_202006_Release_Notes.md (hunk 10 above)
Comment (kannankvs) [reply doc/SONiC_202006_Release_Notes.md]: This was already addressed some time back. Hence, no action.

File: doc/SONiC_202006_Release_Notes.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (henry-ma) [reply doc/SONiC_202006_Release_Notes.md]: @kannankvs instead of mixing them into thermal control, I more incline adding two addtional features - "PSU and FAN LED management" and "PSU, thermal and FAN plugin extension".

"PSU and FAN LED management" - The PSU and FAN LED on switch will be set according to PSU and FAN presence and running status, for example if there is a failure happening to PSU or FAN, the corresponding LED will be set to red. 
On new plugins for fan, thermal and PSU the PSU plugin was extended with voltage, current and power supported, and the fan and thermal plugins were introduced.

**Review by henry-ma — APPROVED** [review]

File: doc/SONiC_202006_Release_Notes.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (kannankvs) [reply doc/SONiC_202006_Release_Notes.md]: Changed. Pls check and confirm.
